homeassistant/components/met/weather.py

```diff
@@ -62,99 +58,77 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     if config.get(CONF_LATITUDE) is None:
         config[CONF_TRACK_HOME] = True
 
-    async_add_entities([MetWeather(config, hass.config.units.is_metric)])
+    await async_setup(hass, config, async_add_entities)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Add a weather entity from a config_entry."""
-    async_add_entities([MetWeather(config_entry.data, hass.config.units.is_metric)])
+    await async_setup(hass, config_entry.data, async_add_entities)
+
+
+async def async_setup(hass, config, async_add_entities):
```

Comment:
the data update coordinator would flow better if placed in __init__ and setup in async_setup_entry there. You can create your own data updater class that stores the MetWeatherData internally and just pass the coordinator around instead via hass.data.

Reply:
added suggestions to help do this. this whole function shouldnt be needed